lib/auth/auth_test.go

```diff
@@ -0,0 +1,63 @@
+package auth
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestNewTokenSuccess(t *testing.T) {
+	f := func(name string, token string, want string) {
+		t.Helper()
+		t.Run(name, func(t *testing.T) {
+			newToken, err := NewToken(token)
+			if err != nil {
+				t.Fatalf("expecting nil error")
+			}
+			got := fmt.Sprintf("%d:%d", newToken.AccountID, newToken.ProjectID)
+			if got != want {
+				t.Errorf("NewToken() got = %v, want %v", newToken, want)
+			}
+		})
+
+	}
+	f("token with accountID and projectID", "1:2", "1:2")
+	f("max uint32 accountID", "4294967295:1", "4294967295:1")
+	f("max uint32 projectID", "1:4294967295", "1:4294967295")
+	f("max uint32 accountID and projectID", "4294967295:4294967295", "4294967295:4294967295")
+}
+
+func TestNewTokenFailure(t *testing.T) {
+	f := func(name string, token string, want string) {
```

Comment:
It looks like the `want` arg is redundant here, since `f` is always called with an empty string as `want` arg.